DependencyInjection/Configuration.php

```diff
@@ -108,4 +109,95 @@ protected function configureClients(ArrayNodeDefinition $root)
                 ->end()
             ->end();
     }
+
+    /**
+     * @param ArrayNodeDefinition $root
+     */
+    protected function configurePlugins(ArrayNodeDefinition $root)
+    {
+        $root->children()
+            ->arrayNode('plugins')
+                ->addDefaultsIfNotSet()
+                ->children()
+
+                    ->arrayNode('authentication')
+                    ->canBeEnabled()
+                        ->children()
+                            ->scalarNode('authentication')->isRequired()->cannotBeEmpty()->end()
+                        ->end()
+                    ->end() // End authentication plugin
+
+                    ->arrayNode('cache')
+                    ->canBeEnabled()
+                    ->addDefaultsIfNotSet()
+                        ->children()
+                            ->scalarNode('cache_pool')->isRequired()->cannotBeEmpty()->end()
+                            ->scalarNode('stream_factory')->cannotBeEmpty()->defaultValue('httplug.stream_factory')->end()
+                            ->arrayNode('config')
+                                ->children()
+                                    ->scalarNode('default_ttl')->defaultNull()->end()
+                                    ->scalarNode('respect_cache_headers')->defaultTrue()->end()
+                                ->end()
+                            ->end()
+                        ->end()
+                    ->end() // End cache plugin
+
+                    ->arrayNode('cookie')
+                    ->canBeEnabled()
+                        ->children()
+                            ->scalarNode('cookie_jar')->isRequired()->cannotBeEmpty()->end()
+                        ->end()
+                    ->end() // End cookie plugin
+
+                    ->arrayNode('decoder')
+                    ->canBeDisabled()
+                    ->addDefaultsIfNotSet()
+                        ->children()
+                            ->scalarNode('use_content_encoding')->defaultTrue()->end()
+                        ->end()
+                    ->end() // End decoder plugin
+
+                    ->arrayNode('history')
+                    ->canBeEnabled()
+                        ->children()
+                            ->scalarNode('journal')->isRequired()->cannotBeEmpty()->end()
```

Comment:
Yeah, This is too much of a replacement for writing a few lines of service configuration yourself. But what does the user expect? That there is config options for all but 2 plugins? 

I think it somehow makes sens to do like this because it is coherent.